File: articles/behavioral-economics-organizational-decision-making/c/organizational_decision_score.c
#include <math.h>
#include <stdio.h>

double logistic(double x) {
    return 1.0 / (1.0 + exp(-x));
}

double approval_probability(
    double expected_payoff,
    double risk,
    double sunk_cost,
    double prestige_value,
    double complexity,
    double overconfidence,
    double short_term_pressure,
    double review_strength,
    double long_horizon_value,
    double long_horizon_weight
) {
    double perceived_value =
        expected_payoff +
        prestige_value * short_term_pressure -
        risk -
        complexity +
        0.9 * sunk_cost +
        0.7 * overconfidence -
        0.8 * review_strength * sunk_cost -
        0.5 * review_strength * overconfidence +
        long_horizon_weight * long_horizon_value;

    return logistic(perceived_value);
}

int main(void) {
    double p = approval_probability(0.14, 0.22, 0.31, 0.20, 0.35, 0.18, 0.7, 0.85, 0.26, 0.60);
    printf("Synthetic approval probability: %.3f\n", p);
    return 0;
}

File: articles/behavioral-economics-organizational-decision-making/rust/organizational_decision_score.rs
fn logistic(x: f64) -> f64 {
    1.0 / (1.0 + (-x).exp())
}

fn approval_probability(
    expected_payoff: f64,
    risk: f64,
    sunk_cost: f64,
    prestige_value: f64,
    complexity: f64,
    overconfidence: f64,
    short_term_pressure: f64,
    review_strength: f64,
    long_horizon_value: f64,
    long_horizon_weight: f64,
) -> f64 {
    let value = expected_payoff
        + prestige_value * short_term_pressure
        - risk
        - complexity
        + 0.9 * sunk_cost
        + 0.7 * overconfidence
        - 0.8 * review_strength * sunk_cost
        - 0.5 * review_strength * overconfidence
        + long_horizon_weight * long_horizon_value;

    logistic(value)
}

fn main() {
    let p = approval_probability(0.14, 0.22, 0.31, 0.20, 0.35, 0.18, 0.70, 0.85, 0.26, 0.60);
    println!("Synthetic approval probability: {:.3}", p);
}
